runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,15 @@
+local M = {}
+
+-- Determine if a *.tf file is TF mud client or terraform
+function M.tf()
+  local number_of_lines = vim.fn.line('$')
+  for line=1,number_of_lines do
+    local current_line = vim.fn.trim(vim.fn.getline(line))
```

Comment:
See my comments here https://github.com/neovim/neovim/pull/18247#discussion_r857203675

Also instead of `vim.fn.line('$')` use `vim.api.nvim_buf_line_count(bufnr)`.

Reply:
(can introduce a local convenience function at the top -- this'll be called often)

Reply:
@clason Did you mean a convenience function for `getline` only or `getline` *and* `buf_line_count`?

Reply:
One for each (possibly including the `trim` in the former, see my comment on the _other_ PR.